### ISSUE TITLE
'winetricks sandbox' seems incompatible with staging-4.19

### ISSUE THREAD

**Issue by tjackson** [Winetricks/winetricks#1425]
I mistakenly contacted the Wine bugzilla first, here: https://bugs.winehq.org/show_bug.cgi?id=48114

It seems 'wine sandbox' causes some problems with staging-4.20, including the inability to run the new Mono installer.

tod@Arch:~$ WINEPREFIX=/home/tod/Downloads/test WINEARCH=win32 WINEDLLOVERRIDES='winemenubuilder.exe=d' winetricks -q sandbox
Using winetricks 20190912-next - sha256sum: 8e423d9f813f9162bf01e7488fb392cf38a0f36dfdf4f2a940bd270acbc7e880 with wine-4.20 (Staging) and WINEARCH=win32
Executing w_do_call sandbox
Executing load_sandbox 
Executing wine regedit /S /D HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Explorer\Desktop\Namespace\{9D20AAE8-0625-44B0-9CA7-71889C2254D9}
wine: could not open working directory L"unix\\home\\tod\\", starting in the Windows directory.
Executing w_do_call isolate_home
Executing load_isolate_home 
Executing cd /home/tod/Downloads/test/drive_c/users/tod
Executing cd /home/tod

Afterwards the working directory message is spammed quite a lot, such as when regsvr32 runs. I'm not sure how much it breaks other things but it's out of my understanding what the problem is.

**Comment by tjackson** [comment]
Sorry, I meant 'winetricks sandbox', of course.

**Comment by Kreyren** [comment]
What is the expected output?

**Comment by tjackson** [comment]
Upon further testing this "could not open" message happens in 4.19 too, but not in 4.18.

Expected output would be the lack of the message:
wine: could not open working directory L"unix\\home\\tod\\Downloads\\", starting in the Windows directory.

**Comment by Kreyren** [comment]
> wine: could not open working directory L"unix\home\tod\", starting in the Windows directory.

assuming current code being:
```sh
#----------------------------------------------------------------

w_metadata sandbox settings \
    title_uk="Пісочниця wineprefix - видалити посилання до HOME" \
    title="Sandbox the wineprefix - remove links to \$HOME"

load_sandbox()
{
    w_skip_windows sandbox && return

    # Unmap drive Z
    rm -f "$WINEPREFIX/dosdevices/z:"

    # Disable unixfs
    # Unfortunately, when you run with a different version of Wine, Wine will recreate this key.
    # See https://bugs.winehq.org/show_bug.cgi?id=22450
    w_try_regedit /D "HKEY_LOCAL_MACHINE\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Desktop\\Namespace\\{9D20AAE8-0625-44B0-9CA7-71889C2254D9}"

    w_call isolate_home
}
```

And assuming that `Executing w_do_call isolate_home` is called after it seems to trigger for `w_try_regedit /D "HKEY_LOCAL_MACHINE\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Desktop\\Namespace\\{9D20AAE8-0625-44B0-9CA7-71889C2254D9}"` which i'm not sure why would it?

Provide output from `winetricks --version`

**Comment by tjackson** [comment]
winetricks --version
20190912-next - sha256sum: 8e423d9f813f9162bf01e7488fb392cf38a0f36dfdf4f2a940bd270acbc7e880

**Comment by Kreyren** [comment]
> 20190912-next - sha256sum: 8e423d9f813f9162bf01e7488fb392cf38a0f36dfdf4f2a940bd270acbc7e880

I can't find that checksum on github https://github.com/Winetricks/winetricks/commit/8e423d9f813f9162bf01e7488fb392cf38a0f36dfdf4f2a940bd270acbc7e880 can you verify that the affected function is same as provided? (load_sandbox() in this case)

**Comment by tjackson** [comment]
Yeah, it looks exactly the same. I also just cloned it and ran ./winetricks/src/winetricks -q sandbox with the same unwanted result.

**Comment by tjackson** [comment]
tod@Arch:~$ ./Downloads/winetricks/src/winetricks --version
20190912-next - sha256sum: 8e423d9f813f9162bf01e7488fb392cf38a0f36dfdf4f2a940bd270acbc7e880

**Comment by tjackson** [comment]
w_call isolate_home doesn't cause the problem so that registry change breaks something I guess.

**Comment by Kreyren** [comment]
I seem to have the same issue on my end:

```console
demo@mx1:~/Desktop
$ rm -r ~/test ; /usr/src/kreytricks/src/winetricks sandbox
------------------------------------------------------
You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Using winetricks 20190912-next - sha256sum: 192dadb0ef20e6925552c6ac6e43b5ea4956ee1ff401948686ca929b54f5660e with wine-4.19 (Staging) and WINEARCH=win64
Executing w_do_call sandbox
Executing load_sandbox 
Executing wine regedit /D HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Explorer\Desktop\Namespace\{9D20AAE8-0625-44B0-9CA7-71889C2254D9}
wine: could not open working directory L"unix\\home\\demo\\Desktop\\", starting in the Windows directory.
Executing wine64 regedit /D HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Explorer\Desktop\Namespace\{9D20AAE8-0625-44B0-9CA7-71889C2254D9}
wine: could not open working directory L"unix\\home\\demo\\Desktop\\", starting in the Windows directory.
Executing w_do_call isolate_home
Executing load_isolate_home 
Executing cd /home/demo/test/drive_c/users/demo
Executing cd /home/demo/Desktop
```

**Comment by Kreyren** [comment]
I'm not sure how to fix this excluding regexing `$WINEPREFIX/system.reg` -> Passing

**Comment by austin987** [comment]
Looks like this changed upstream in:
```
austin@laptop ~/wine-sandbox ((no branch, bisect started on master)) $ git bisect good
305b5a44c57015ff6b981f83c985c85b223058e8 is the first bad commit
commit 305b5a44c57015ff6b981f83c985c85b223058e8
Author: Alexandre Julliard <julliard@winehq.org>
Date:   Tue Oct 22 09:58:32 2019 +0200

    kernel32: Move current directory initialization to ntdll.
    
    Signed-off-by: Alexandre Julliard <julliard@winehq.org>

:040000 040000 6236519c71e62bcd8ead9975e4697088d204c996 44c015d73ae32614f641adfd79988f90978f2688 M	dlls
```

upsream bugzilla is down at the moment, so I can't check the other bug/report it, will later.

**Comment by tjackson** [comment]
I'm pretty sure this won't be an issue anymore in 4.21 based on testing from Wine git just now (with and without staging patches) but I'll let you know once 4.21 is released. Can't say which exact commit fixed it however.

**Comment by tjackson** [comment]
Actually, forget I said anything. Staging is broken for the 32-bit build right now. 000d:err:module:LdrInitializeThunk "kernelbase.dll" failed to initialize, aborting
So I'll try looking into this more when the whole multilib part is working/building correctly. Sorry for the noise.

**Comment by tjackson** [comment]
It seems to be all but cosmetically fixed by https://source.winehq.org/git/wine.git/?a=commit;h=f2060812418bc049105c2a30ceb039ee6d5f0fd7

I can still trigger the message which I'll show with the console output I attached.
[test.txt](https://github.com/Winetricks/winetricks/files/3903416/test.txt)

In regular use it doesn't appear anymore and that one weird instance with my erroneous 'wine start' seems harmless. Thanks for the bisects, and sorry if I stepped on any toes in the bug report.
